v2/contribute/decisions/session/0031-internal-prop-handling-in-flat-access-tokens.mdx

```diff
@@ -0,0 +1,62 @@
+---
+id: "0031"
+title: Do not add internal only claims to the front-token
+hide_title: true
+---
+
+import DecisionHeader from "/src/components/decisionLogs/DecisionHeader";
+import ArgumentPro from "/src/components/decisionLogs/ArgumentPro";
+import ArgumentCon from "/src/components/decisionLogs/ArgumentCon";
+import ArgumentNeut from "/src/components/decisionLogs/ArgumentNeut";
+
+# Do not add internal only claims to the front-token
+
+<DecisionHeader
+  status="proposed"
+  lastUpdate="2023-01-22"
+  created="2023-01-22"
+  deciders={["rishabhpoddar", "porcellus"]}
+  proposedBy={["porcellus"]}
+/>
+
+## Context and Problem Statement
+
+When flattening the access token structure and making it into a standard JWT, we have to decide if it's safe to add all claims (including internally used ones) into the front-token, or not.
+
+## Considered Options
+
+- Hide internal-only claims
+- Hide all claims not added by the user
+- Expose all claims
+
+## Decision Outcome
+
+We should hide internal-only claims
+
+- We don't want to confuse our users
+- They are never used in applications
+
+We decided to hide: `refreshTokenHash1`, `parentRefreshTokenHash1`, because they do not make sense for the application and are never used as a part of CDI or FDI. 
+We can expose `sub`, `exp` and `iat`, because they have standard names and carry potentially useful information that we expose anyway.
+Exposing the `sessionHandle` should not cause any issues and could be useful in very rare cases on the frontend to detect session changes. This is only used on the CDI, while getting session information and updating sessionData in the database (not in access token). As a side-note, using the session handle to call `updateSessionData` doesn't seem to be documented.
+
+### Hide internal-only claims
+
+<ArgumentPro> Hides claims that would be useless to the application </ArgumentPro>
+<ArgumentPro> Exposes the standard claims (people have tried to get them through `getAccessTokenPayload` before) </ArgumentPro>
+<ArgumentCon> Changes the current behaviour/interface </ArgumentCon>
+
+### Hide all claims not added by the user
+
+<ArgumentPro> Matches the current behaviour </ArgumentPro>
+<ArgumentPro> Hides all claims that could be confusing </ArgumentPro>
+<ArgumentCon> People have been looking for the userId claim </ArgumentCon>
+<ArgumentCon> We are hiding otherwise standard claims </ArgumentCon>
+
+### Expose all claims 
+
+<ArgumentPro> Front-token is very straght-forward </ArgumentPro>
+<ArgumentCon> Changes the current behaviour/interface </ArgumentCon>
+<ArgumentPro> Exposes the standard claims (people have tried to use them through getAccessTokenPayload before) </ArgumentPro>
+<ArgumentCon> Could be confusing, we'd need to document each internal claim (and/or move them into a separate `st_internal` claim) </ArgumentCon>
```

Comment:
not really.. we don't need to document this. If someone asks, we can just answer somewhere. Not a big deal.